krun/scheduler.py

```diff
@@ -309,16 +309,54 @@ def _write_new_manifest(self, config):
             for item in manifest:
                 fh.write("%s\n" % item)
 
+    def next_exec_key_index(self):
+        """Returns the sequential process execution index into the ordered list
+        of all process executions sharing the same 'bench:vm:variant' key.
+
+        Although this could have been done at `_parse()` time, it would require
+        a (variable sized) `dict` since we don't know which key we will be
+        counting for until we find the first outstanding (O) record.
+
+        Instead, this method does a pass over the manifest searching for
+        records whose key is `self.next_exec_key`.
```

Comment:
I like this very clear explanation!

Reply:
Thanks. It took a while to get the wording right on this one!